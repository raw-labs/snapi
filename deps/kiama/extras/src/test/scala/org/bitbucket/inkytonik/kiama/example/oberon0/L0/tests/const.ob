MODULE const;

CONST
    aconst1 = 1;
    aconst2 = aconst1 + 100;
    aconst3 = 400 DIV aconst2;
    aconst4 = noconst + 1;
    aconst5 = 1 DIV 0;
    aconst6 = 2 MOD (3 DIV 4);

VAR
    avar : INTEGER;

BEGIN
    avar := aconst1 * aconst2 MOD aconst3 - aconst4;
    aconst1 := 42
END const.
