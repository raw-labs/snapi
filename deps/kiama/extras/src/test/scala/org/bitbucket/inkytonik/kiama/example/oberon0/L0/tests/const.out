* _pp.ob
MODULE const;

    CONST
        aconst1 = 1;
        aconst2 = aconst1 + 100;
        aconst3 = 400 DIV aconst2;
        aconst4 = noconst + 1;
        aconst5 = 1 DIV 0;
        aconst6 = 2 MOD (3 DIV 4);

    VAR
        avar : INTEGER;

BEGIN
    avar := aconst1 * aconst2 MOD aconst3 - aconst4;
    aconst1 := 42
END const.
* errors
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/const.ob:7:15:error: noconst is not declared
    aconst4 = noconst + 1;
              ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/const.ob:8:21:error: division by zero in constant expression
    aconst5 = 1 DIV 0;
                    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/const.ob:9:22:error: modulus by zero in constant expression
    aconst6 = 2 MOD (3 DIV 4);
                     ^
[2]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L0/tests/const.ob:16:5:error: illegal assignment
[2]    aconst1 := 42
[2]    ^
